### DISCUSSION TITLE
Consider a lightweight option for path-only functionality

### DISCUSSION THREAD

**Discussion by mattwigham** [ElMassimo/js_from_routes#54] [Ideas]
Thanks @ElMassimo for such a great lib. 

For my project, I needed a way to generate paths in JS, but I didn't need the client features since I already had those elsewhere. 

I was able to create a more lightweight version for path-only functionality using the custom client library below, but I thought you might consider adding something like this as an option, or even the default, and let people opt-in to the additional client features if they need them.

```js
import { formatUrl } from "@js-from-routes/core";

export const definePathHelper = (_, path) => {
  return (params) => formatUrl(path, params);
};
```

which simplifies usage to...

```js
products.show({ id: 123 })
```

**Comment by ElMassimo** [comment]
Hi Matt!

This usage seems like something that would be nice to cover in the docs, and maybe have an example repo.

I always recommend using [custom code](https://js-from-routes.netlify.app/guide/codegen.html#using-your-own-code), as that makes it easier to build upon the provided client libraries.

```ruby
config.client_library = '~/services/api'
```

so your solution is what I would typically use if I wanted that behavior.

When using TypeScript, you can leverage `UrlOptions`, provided by the library:

```ts
// app/frontend/services/api.ts
import { formatUrl, type UrlOptions } from '@js-from-routes/core'

export function definePathHelper (_method: string, pathTemplate: string) {
  return (options?: UrlOptions) => formatUrl(pathTemplate, options)
}
```

That will provide better type-checking, or autocompletion if mostly using JS.
